### PR TITLE
Downgrade warning in __init__ to logger.debug

### PR DESCRIPTION
**Pull request by mrocklin**
We may not want to be this vocal/scary with non-expert users

### CODE REVIEW

File: ucp/__init__.py [hunk 1]
```diff
@@ -12,13 +12,13 @@
...
+logger = logging.getLogger("ucx")
+
 # Notice, if we have to update environment variables
 # we need to do it before importing UCX
 if os.environ.get("UCX_MEMTYPE_CACHE", "") != "n":
     # See <https://github.com/openucx/ucx/wiki/NVIDIA-GPU-Support#known-issues>
-    warnings.warn(
-        "Setting env UCX_MEMTYPE_CACHE=n, which is required by UCX", UCXWarning
-    )
+    logger.debug("Setting env UCX_MEMTYPE_CACHE=n, which is required by UCX")
```
Comment (mrocklin) [inline ucp/__init__.py]: I'm also curious about setting the default logging level below, and using the global logger.  @madsbk git blame points to you.  Can I ask you for your thoughts on the lines just below these?

Comment (madsbk) [reply ucp/__init__.py]: I haven't given it much thought :)
You are more then welcome to change it.

**Comment by quasiben** [comment]
@jakirkham can you review and merge ?

**Comment by jakirkham** [comment]
Yeah was about to ask if we are ready to merge this. Was unclear if we were waiting for more changes. Happy to merge as-is and iterate 🙂

**Comment by jakirkham** [comment]
Matt said it was ok to merge offline. So merging...

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**